service_configuration_lib/spark_config.py

```diff
@@ -154,6 +165,36 @@ def get_aws_credentials(
     )
 
 
+def assume_aws_role_web_identity():
+    """
+    Checks that a web identity token is available, and if it is,
+    get an aws session and return a credentials dictionary
+    """
+    if 'AWS_WEB_IDENTITY_TOKEN_FILE' not in os.environ or 'AWS_ROLE_ARN' not in os.environ:
+        return {}
+    with open(os.environ['AWS_WEB_IDENTITY_TOKEN_FILE']) as token_file:
+        token = token_file.read()
+    role_arn = os.environ['AWS_ROLE_ARN']
+    timestamp = int(time.time())
+    client = boto3.client('sts')
+    try:
+        resp = client.assume_role_with_web_identity(
+            RoleArn=role_arn,
+            RoleSessionName=f'SparkRun-{timestamp}',
+            WebIdentityToken=token,
+            DurationSeconds=43200,
+        )
+    except botocore.exceptions.ClientError:
+        resp = client.assume_role_with_web_identity(
+            RoleArn=role_arn,
+            RoleSessionName=f'SparkRun-{timestamp}',
+            WebIdentityToken=token,
+            DurationSeconds=3600,
+        )
+        log.warning(f"Session duration for {role_arn} is only 1 hour due to the role's maximum session duration")
```

Comment:
You could consider just making the durationSeconds a parameter for the function, and letting the failure bubble up if an invalid value is chosen. In theory, a role's max session could be as low as 15 minutes.